examples/eSealing/src/lib.rs

```diff
@@ -0,0 +1,444 @@
+//! A contract implementing an eSealing service.
+//!
+//! Only accounts can register a file hash. During the registration, the
+//! timestamp is recorded together with the witness (sender_account) that
+//! registered the file hash.
+//!
+//! Every account can only register each file hash
+//! once. This is because we want to record the time when the witness
+//! (sender_account) first had access to that file. Re-registering a file hash
+//! by the same account would not give us any additional information.
+//!
+//! But: A second account can register a file hash that was already previously
+//! registered. This second account will be added as another witness together
+//! with the timestamp when the second account called the `register_file`
+//! function. We know that the second account is most likely now also in
+//! possession of the file if it has used this eSealing service via the
+//! front-end. But this is not guaranteed since the second account could
+//! register a random file hash it found in the events by interacting with the
+//! smart contract directly (without using the front-end).
+use concordium_std::*;
+use std::ops::Deref;
+
+/// The different errors the contract can produce.
+#[derive(Serialize, Debug, PartialEq, Eq, Reject, SchemaType)]
+enum ContractError {
+    /// Failed parsing the parameter.
+    #[from(ParseError)]
+    ParseParams,
+    /// Failed logging: Log is full.
+    LogFull,
+    /// Failed logging: Log is malformed.
+    LogMalformed,
+    /// Only accounts can register a file hash.
+    OnlyAccount,
+    /// Every account can only register each file hash once.
+    AlreadyRecorded,
+}
+
+/// Mapping the logging errors to ContractError.
+impl From<LogError> for ContractError {
+    fn from(le: LogError) -> Self {
+        match le {
+            LogError::Full => Self::LogFull,
+            LogError::Malformed => Self::LogMalformed,
+        }
+    }
+}
+
+/// The state tracked for each file.
+#[derive(Serial, DeserialWithState, Deletable, StateClone)]
+#[concordium(state_parameter = "S")]
+struct FileState<S> {
+    /// The timestamps when this file hash was recorded.
+    timestamps:    Vec<Timestamp>,
+    /// The witnesses (sender_accounts) that recorded this file hash.
+    witnesses:     Vec<AccountAddress>,
```

Comment:
I am not sure why we need this vector. Is it to order by increasing timestamp?

Is the length of timestamps always the same as witnesses?